doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -8,16 +8,154 @@ NEP 35 — Array Creation Dispatching With __array_function__
 :Status: Draft
 :Type: Standards Track
 :Created: 2019-10-15
-:Updated: 2020-08-06
+:Updated: 2020-08-17
 :Resolution:
 
 Abstract
 --------
 
 We propose the introduction of a new keyword argument ``like=`` to all array
-creation functions to permit dispatching of such functions by the
-``__array_function__`` protocol, addressing one of the protocol shortcomings,
-as described by NEP-18 [1]_.
+creation functions, this argument permits the creation of an array based on
+a non-NumPy reference array passed via that argument, resulting in an array
+defined by the downstream library implementing that type, which also implements
+the ``__array_function__`` protocol. With this we address one of that
+protocol's shortcomings, as described by NEP 18 [1]_.
+
+Motivation and Scope
+--------------------
+
+Many are the libraries implementing the NumPy API, such as Dask for graph
+computing, CuPy for GPGPU computing, xarray for N-D labeled arrays, etc. All
+the libraries mentioned have yet another thing in common: they have also adopted
+the ``__array_function__`` protocol. The protocol defines a mechanism allowing a
+user to directly use the NumPy API as a dispatcher based on the input array
+type. In essence, dispatching means users are able to pass a downstream array,
+such as a Dask array, directly to one of NumPy's compute functions, and NumPy
+will be able to automatically recognize that and send the work back to Dask's
+implementation of that function, which will define the return value. For
+example:
+
+.. code:: python
+
+    x = dask.array.arange(5)    # Creates dask.array
+    np.sum(a)                   # Returns dask.array
+
+Note above how we called Dask's implementation of ``sum`` via the NumPy
+namespace by calling ``np.sum``, and the same would apply if we had a CuPy
+array or any other array from a library that adopts ``__array_function__``.
+This allows writing code that is agnostic to the implementation library, thus
+users can write their code once and still be able to use different array
+implementations according to their needs.
+
+Unfortunately, ``__array_function__`` has limitations, one of them being array
```

Comment:
Ah this was my initial misunderstanding. How about this? 

```
Obviously, having a protocol in-place is useful if the arrays are created
somewhere else and let NumPy handle them. But still these arrays have
to be started in their native library and brought back. Instead if it was
actually possible to create these objects through NumPy API then there
would be an almost full experience all using NumPy syntax. For example, 
say we have some CuPy object ``C`` , and want a similar CuPy array with all ones
 in it. We could still write 


.. code:: python
    x = cupy.array.ones(....)  # I don't know cupy syntax for this.

But instead the better way would be `x = np.ones([100, 25], like=C)` then
magically `x` will also be a CuPy array, learning from the type of ``C``.

Reply:
Reused your structure but changed the wording a bit, let me know if it's clear enough like that or if there's room for improvement still.